src/sentry/migrations/0405_monitor_cleanup.py

```diff
@@ -0,0 +1,67 @@
+# Generated by Django 2.2.28 on 2023-04-04 16:36
+
+from django.db import migrations
+
+from sentry.monitors.models import MonitorStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def clean_up_monitors(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+    Organization = apps.get_model("sentry", "Organization")
+    Project = apps.get_model("sentry", "Project")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        Monitor.objects.filter(monitorenvironment__isnull=True)
+        .exclude(status__in=[MonitorStatus.PENDING_DELETION, MonitorStatus.DELETION_IN_PROGRESS])
+        .values_list(
+            "id",
+            "organization_id",
+            "project_id",
+        ),
+        result_value_getter=lambda item: item[0],
+    )
+
+    monitors_to_clean_up = []
+
+    for monitor_id, organization_id, project_id in queryset:
+        try:
+            Organization.objects.get(id=organization_id)
+            Project.objects.get(id=project_id)
+        except (Organization.DoesNotExist, Project.DoesNotExist):
+            monitors_to_clean_up.append(monitor_id)
+
+    Monitor.objects.filter(id__in=monitors_to_clean_up).delete()
```

Comment:
do we want exception handling here? or would rather error out so we can roll back quickly?

Reply:
or should I be scheduling the deletion?
https://github.com/getsentry/sentry/blob/974a65c00636d0cca8b2421f884e2f15c13063d9/src/sentry/monitors/endpoints/organization_monitor_details.py#L142-L161